### PR TITLE
feat(api/task): add task API; fix cross-user resource fetching

### PR DESCRIPTION
**Pull request by brettstack**
*Issue #, if available:*

*Description of changes:*

# Checklist

- [ ] Tests have been added and are passing
- [ ] Documentation has been updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/api/controllers/disease.js [hunk 1]
```diff
@@ -5,8 +5,6 @@ import { log } from '../utils/logger'
 export async function createDisease({
   disease,
 }) {
-  if (!disease) throw new Error('disease is required')
```
Comment (sanathkr) [inline packages/api/controllers/disease.js:8]: why remove these validations?

File: packages/api/controllers/project.js [hunk 2]
```diff
@@ -3,17 +3,15 @@ import { generateId } from '../utils/id'
...
 
 export async function createProject({
+  userId,
   programId,
   project,
 }) {
-  if (!programId) throw new Error('programId is required')
-  if (!project) throw new Error('project is required')
-
   const id = generateId()
```
Comment (sanathkr) [inline packages/api/controllers/project.js:10]: `id` is unused

File: packages/api/controllers/project.js [hunk 3]
```diff
@@ -3,17 +3,15 @@ import { generateId } from '../utils/id'
...
   const item = {
     ...project,
+    programId: `${userId}#${programId}`,
```
Comment (sanathkr) [inline packages/api/controllers/project.js:13]: - should we call this the partition key explicitly so we don't assume this is the real `programId` - same comment applies to other places in this doc.
- Can we create a method to return the partition key so we keep it DRY

File: packages/api/controllers/task.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+export async function createTask({
+  userId,
+  projectId,
+  task,
+}) {
+  const taskId = generateId()
+  const item = {
+    ...task,
+    projectId: `${userId}#${projectId}`,
```
Comment (sanathkr) [inline packages/api/controllers/task.js:13]: same comment as above

File: packages/api/functions/express/routes/program.js [hunk 5]
```diff
@@ -20,20 +20,24 @@ programRouter.post('/', wrapAsync(async (req, res) => {
 
 programRouter.put('/:programId', wrapAsync(async (req, res) => {
   const { programId } = req.params
+  const userId = req.cognitoUser.id
```
Comment (sanathkr) [inline packages/api/functions/express/routes/program.js:23]: what happens if thhe API is called unauthenticated?

File: packages/api/models/Project.js [hunk 6]
```diff
@@ -3,6 +3,7 @@ import { dynamoDbDocumentClient } from '../dynamodb-init'
 
 const ProjectTable = new Table({
   name: process.env.PROJECT_TABLE,
+  // userId#programId
```
Comment (sanathkr) [inline packages/api/models/Project.js:6]: remove

File: packages/api/models/Task.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
+import { Table, Entity } from 'dynamodb-toolbox'
+import { dynamoDbDocumentClient } from '../dynamodb-init'
+
+const TaskTable = new Table({
+  name: process.env.TASK_TABLE,
+  // userId#projectId
```
Comment (sanathkr) [inline packages/api/models/Task.js:6]: remove - this can quickly go outdated

File: packages/ui/pages/_app.tsx [hunk 8]
```diff
@@ -96,23 +88,23 @@ const federated = {
...
+    <ConfirmSignUpRedirectToSignIn override="ConfirmSignUp" />,
+    <ForgotPassword />,
+    <RequireNewPassword />,
+    <Loading />,
+    <Greetings />,
+  ],
+});
 
-export default MyApp;
+// export default MyAppWrapper;
```
Comment (sanathkr) [inline packages/ui/pages/_app.tsx:110]: remove?

**Review by sanathkr — CHANGES_REQUESTED** [review]
This looks great - ssuper easy to read and understand!

File: packages/api/controllers/disease.js (hunk 1 above)
Comment (brettstack) [reply packages/api/controllers/disease.js:8]: Will add back in

File: packages/api/controllers/project.js (hunk 3 above)
Comment (brettstack) [reply packages/api/controllers/project.js:13]: Discussed on call. We'll call it `pk` and also call `id` `sk`

File: packages/api/functions/express/routes/program.js (hunk 5 above)
Comment (brettstack) [reply packages/api/functions/express/routes/program.js:23]: API Gateway Cognito Authorizer prevents it from reaching Lambda

File: packages/api/models/Project.js (hunk 6 above)
Comment (brettstack) [reply packages/api/models/Project.js:6]: I'll update this to something like `// NOTE: The partition key looks like userId#programId`

**Review by sanathkr — APPROVED** [review]

**Pull request merged**